### PR TITLE
Introduce Run3 (post-LS2) workflow

### PR DESCRIPTION
**Pull request by kpedro88**
This PR introduces a Run3 workflow for the post-LS2 detector (including GE1/1 and HB upgrades, along with Castor removal). The Run2_2018 GT is used for now. Because HCAL conditions don't exist yet for the post-LS2 geometry, the hardcode conditions are used (via the new modifier setup introduced in #19712).

A few modifications are made to the GEM geometry based on phase2 developments (discussed with experts). In addition, I had to fix some logic in a muon DQM analyzer to prevent a crash; with GE1/1 included, we apparently get some muons that fail all the conditions `isGlobalMuon()`, `isTrackerMuon()`, `isRPCMuon()`, `isStandAloneMuon()`. For now, those muons are just skipped. Muon experts may want to improve this handling in the future.

attn: @calabria, @bsunanda, @ianna

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
relval failure was expected, seems to be due to pixel Phase 1 DQM changes (ongoing investigation, discussion in #19811)
no idea about the material budget failure, the log doesn't indicate anything.

**Comment by kpedro88** [comment]
(NB: this ran fine in `CMSSW_9_3_0_pre2`)

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by kpedro88** [comment]
@fabozzi please sign

**Comment by kpedro88** [comment]
@fabozzi ping

**Comment by fabozzi** [comment]
+1

File: Configuration/StandardSequences/python/Reconstruction_cff.py [hunk 1]
```diff
@@ -50,16 +50,16 @@
 localreco = cms.Sequence(bunchSpacingProducer+trackerlocalreco+muonlocalreco+calolocalreco+castorreco)
 localreco_HcalNZS = cms.Sequence(bunchSpacingProducer+trackerlocalreco+muonlocalreco+calolocalrecoNZS+castorreco)
 
-_phase2_localreco = localreco.copyAndExclude([castorreco])
-_phase2_localreco_HcalNZS = localreco_HcalNZS.copyAndExclude([castorreco])
-from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
-phase2_common.toReplaceWith(localreco, _phase2_localreco)
-phase2_common.toReplaceWith(localreco_HcalNZS, _phase2_localreco_HcalNZS)
+_run3_localreco = localreco.copyAndExclude([castorreco])
```
Comment (davidlange6) [inline Configuration/StandardSequences/python/Reconstruction_cff.py]: Hi @kpedro88 - was this choice based on some discussion?

File: Geometry/MuonSimData/data/2019/muonProdCuts.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0"?>
```
Comment (davidlange6) [inline Geometry/MuonSimData/data/2019/muonProdCuts.xml]: what is the idea behind copying this file? Does it reflect a detector change?

File: Configuration/StandardSequences/python/Reconstruction_cff.py (hunk 1 above)
Comment (kpedro88) [inline Configuration/StandardSequences/python/Reconstruction_cff.py]: The post-LS2 beampipe requires CASTOR to be removed.

File: Geometry/MuonSimData/data/2019/muonProdCuts.xml (hunk 2 above)
Comment (kpedro88) [inline Geometry/MuonSimData/data/2019/muonProdCuts.xml]: L37 is different to reflect the addition of GEMS:
```
 <PartSelector path="//GE.*"/>
```

File: Configuration/StandardSequences/python/Reconstruction_cff.py (hunk 1 above)
Comment (slava77) [inline Configuration/StandardSequences/python/Reconstruction_cff.py]: @hvanhaev 
please confirm that there will be no running with castor after LS2.
Thank you.

**Comment by davidlange6** [comment]
merge

**Comment by Dr15Jones** [comment]
It appears this pull request added workflow 11626 to the nightly IB and that workflow is failing:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_aarch64_gcc700/CMSSW_9_3_X_2017-08-09-1100/pyRelValMatrixLogs/run/11626.0_QCD_Pt_600_800_13+QCD_Pt_600_800_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+ALCAFull_2019+HARVESTFull_2019/step2_QCD_Pt_600_800_13+QCD_Pt_600_800_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+ALCAFull_2019+HARVESTFull_2019.log

the exception being thrown is some variation of a request out of bounds
```
----- Begin Fatal Exception 09-Aug-2017 13:34:49 CEST-----------------------
An exception of category 'Out of Range' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'digitisation_step'
   [2] Calling method for module HcalTrigPrimDigiProducer/'simHcalTriggerPrimitiveDigis'
Exception Message:
LUT has 1024 entries for (HcalTrigTower v0: -3,50) but 1596 was requested.
----- End Fatal Exception -------------------------------------------------
```

**Comment by kpedro88** [comment]
@Dr15Jones ah, the short matrix only runs 11624... I'll investigate

**Comment by kpedro88** [comment]
@Dr15Jones see #20104

**Comment by hvanhaev** [comment]
Dear Slava,

Indeed, at the moment there are no data taking plans with CASTOR after LS2.

Cheers,
Hans


On 8 Aug 2017, at 17:07, Slava Krutelyov <notifications@github.com<mailto:notifications@github.com>> wrote:


@slava77 commented on this pull request.

________________________________

In Configuration/StandardSequences/python/Reconstruction_cff.py<https://github.com/cms-sw/cmssw/pull/19889#discussion_r131940516>:

> @@ -50,16 +50,16 @@
 localreco = cms.Sequence(bunchSpacingProducer+trackerlocalreco+muonlocalreco+calolocalreco+castorreco)
 localreco_HcalNZS = cms.Sequence(bunchSpacingProducer+trackerlocalreco+muonlocalreco+calolocalrecoNZS+castorreco)

-_phase2_localreco = localreco.copyAndExclude([castorreco])
-_phase2_localreco_HcalNZS = localreco_HcalNZS.copyAndExclude([castorreco])
-from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
-phase2_common.toReplaceWith(localreco, _phase2_localreco)
-phase2_common.toReplaceWith(localreco_HcalNZS, _phase2_localreco_HcalNZS)
+_run3_localreco = localreco.copyAndExclude([castorreco])


@hvanhaev<https://github.com/hvanhaev>
please confirm that there will be no running with castor after LS2.
Thank you.

**Pull request merged**